Sources/NIOHTTP2/NGHTTP2Session.swift

```diff
@@ -285,7 +285,7 @@ fileprivate struct StreamManager {
     // Discard old and unnecessary streams.
     private mutating func purgeOldStreams() {
         while self.streamMap.count >= maxSize {
-            let lowestStreamID = self.streamMap.filter { $0.value.active }.keys.sorted().first { $0 != 0 && $0 != Int32.max }!
+            let lowestStreamID = self.streamMap.filter { !$0.value.active }.keys.sorted().first { $0 != 0 && $0 != Int32.max }!
```

Comment:
I assume the filtered array (and, by extension, `self.streamMap`) will now never be empty?

Reply:
Not quite, but I wanted to separate the other issue out into another patch.

The TL;DR there is that if the sum of both sides SETTINGS_MAX_CONCURRENT_STREAMS is higher than `maxSize`, and there are actually more than `maxSides` valid open active streams, then we'll explode again. That can be fixed by changing the loop conditional above to move the `filter` out of the loop so that we only care if there are too many *closed* streams.

This will all get a bit better when we land #10 and the follow-on framing logic such that I can start work on replacing the nghttp2 state machine. At that stage we'll have a much more complete insight into the connection state and won't have to use this somewhat insane purging construction.